jkube-kit/common/src/test/java/org/eclipse/jkube/kit/common/util/Fabric8HttpUtilTest.java

```diff
@@ -49,7 +50,91 @@ void toFormData_whenDataProvidedAsMap_thenCreateFormDataPayload() throws Unsuppo
   }
 
   @Test
-  void extractAuthenticationChallengeIntoMap_whenWwwHeaderProvided_thenParseDataIntoMap() {
+  void extractAuthenticationChallengeIntoMap_whenEmptyHeaderProvided_thenReturnEmptyMap() throws IOException {
+    // Given
+    Map<String, List<String>> responseHeaders = new HashMap<>();
+    responseHeaders.put("WWW-Authenticate", Collections.singletonList(""));
+    HttpResponse<byte[]> response = new TestHttpResponse<byte[]>(responseHeaders).withCode(HTTP_OK);
+    // When
+    Map<String, String> result = Fabric8HttpUtil.extractAuthenticationChallengeIntoMap(response);
+    // Then
+    assertThat(result).isEmpty();
+  }
+
+  @Test
+  void extractAuthenticationChallengeIntoMap_whenBasicAuthenticationRealm_thenParseDataIntoMap() throws IOException {
+    // Given
+    Map<String, List<String>> responseHeaders = new HashMap<>();
+    responseHeaders.put("WWW-Authenticate", Collections.singletonList("Basic realm=\"Access to the staging site\", charset=\"UTF-8\""));
+    HttpResponse<byte[]> response = new TestHttpResponse<byte[]>(responseHeaders).withCode(HTTP_OK);
+    // When
+    Map<String, String> result = Fabric8HttpUtil.extractAuthenticationChallengeIntoMap(response);
+    // Then
+    assertThat(result)
+        .hasSize(2)
+        .containsEntry("Basic realm", "Access to the staging site")
+        .containsEntry("charset", "UTF-8");
+  }
+
+  @Test
+  void extractAuthenticationChallengeIntoMap_whenMultipleChallenges_thenParseDataIntoMap() throws IOException {
+    // Given
+    Map<String, List<String>> responseHeaders = new HashMap<>();
+    responseHeaders.put("WWW-Authenticate", Collections.singletonList("Newauth realm=\"apps\", type=1, title=\"Login to \"apps\"\", Basic realm=\"simple\""));
```

Comment:
This is what I was referring to in the internal call.
I don't think that this is a valid value, it will probably come with escaped quotations marks, as in `title=\"something \\\"escaped\\\"\"`

The idea would be to reuse something that performs the parsing for us. Not sure if Jackson, XML, or something we can leverage

Reply:
So, for example, OkHttp includes the 
HttpHeadrs.parseChallenges method